app/models/user.rb

```diff
@@ -5,4 +5,8 @@ class User < ApplicationRecord
 
   devise :database_authenticatable, :registerable, :recoverable, :rememberable, :validatable,
     :confirmable, :omniauthable, omniauth_providers: %i[google_oauth2 github]
+
+  def balance
+    transactions.map { |tr| tr.amount.to_i }.sum
+  end
```

Comment:
@marderer `#balance` in User model is a good idea 👍

Reply:
@marderer what is `tr`? if it's a `transition` then let's call it `transition` 😉

Reply:
@marderer why not
```ruby
def balance
  transactions.sum
end
```
instead of
```ruby
def balance
  transactions.map { |tr| tr.amount.to_i }.sum
end
```
?

It's not User responsibility to know how a sum of transitions is calculated, the only thing User should know - is that we need this sum, so `transitions.sum` 
and in `Transition` model:
```ruby
def self.sum
  sum(&:amount)
end
```

Reply:
@marderer do we really need `amount.to_i`? `amount` is numerical, so should be Integer from a box

Reply:
@AleksSenkou >It's not User responsibility to know how a sum of transitions is calculated, the only thing User should know - is that we need this sum, so transitions.sum
Ok, I understood you, but this method can't help me. 

``` ruby
def self.sum
end
```
This class method will allow cal ```Transactions.sum``` and will be equal the amount of all transactions, but I should call, for example ```User.first.transactions.sum```, amount only those transactions that the user added and class method don't work here.

Reply:
@AleksSenkou 
>do we really need amount.to_i? amount is numerical, so should be Integer from a box

Yes, we really should use ```to_i```, because amount isn't integer. It is a string with numerical validation, but this validation doesn't convert string to integer, it checks that only numbers are written to the string.

Reply:
@AleksSenkou 
[This solution is suitable?](https://github.com/marderer/personal-finance/pull/25/commits/b3d01414237e32e1874f04a2f228e3dab5942e01)

Reply:
@marderer yeah, it's a suitable solution 👍